### PR TITLE
Fix mypy for plugins registry

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fix mypy for plugins registry

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

File: pytorch_lightning/plugins/plugins_registry.py [hunk 1]
```diff
@@ -102,6 +102,9 @@ def get(self, name: str) -> Any:
             data = self[name]
             return data["plugin"](**data["init_params"])
 
+        if default is not None:
+            return default
+
```
Comment (ananthsub) [inline pytorch_lightning/plugins/plugins_registry.py:107]: why do we need the default?

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 2]
```diff
@@ -332,3 +332,12 @@ def predict_step(self, *args, **kwargs):
...
             self.model.require_backward_grad_sync = True
+
+    @classmethod
+    def register_plugins(cls, plugin_registry: Dict) -> None:
+        plugin_registry.register(
+            "ddp_with_find_unused_parameters_false",
+            cls,
+            description="DDP Plugin with `find_unused_parameters` as False",
+            find_unused_parameters=False
+        )
```
Comment (ananthsub) [inline pytorch_lightning/plugins/training_type/ddp.py:343]: i'd prefer splitting this out to a separate PR so this PR fixes only the typing. wdyt?

File: pytorch_lightning/plugins/plugins_registry.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/plugins/plugins_registry.py:107]: That's the signature for the `get` method.
Or else, it raises a concern
```
pytorch_lightning/plugins/plugins_registry.py:93: error: Signature of "get" incompatible with supertype "Mapping"  [override]
```

File: pytorch_lightning/plugins/training_type/ddp.py (hunk 2 above)
Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/ddp.py:343]: Actually, I found an issue with the Registry. Will do a separate PR for the fix and this.

**Review by ananthsub — APPROVED** [review]
Thanks!

**Review by awaelchli — APPROVED** [review]

**Pull request merged**